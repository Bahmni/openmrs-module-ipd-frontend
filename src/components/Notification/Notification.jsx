import React from "react";
import PropTypes from "prop-types";
import { NotificationCarbon } from "bahmni-carbon-ui";
import { FormattedMessage } from "react-intl";
import { I18nProvider } from "../../features/i18n/I18nProvider";
import "./Notification.scss";

export default function Notification(props) {
  const { hostData, hostApi } = props;
  const getMessage = () => {
    switch (hostData?.notificationKind) {
      case "success":
        return (
          <I18nProvider>
            <FormattedMessage id={hostData.messageId} />
          </I18nProvider>
        );
      case "warning":
        return (
          <I18nProvider>
            <FormattedMessage id={hostData.messageId} />
          </I18nProvider>
        );
    }
  };
  const title = getMessage();

  return (
    <NotificationCarbon
      className="notification"
      showMessage={true}
      title={title}
      kind={hostData?.notificationKind}
      onClose={() => {
        hostApi?.onClose();
      }}
      hideCloseButton={true}
    />
  );
}

Notification.propTypes = {
  hostData: PropTypes.shape({
    notificationKind: PropTypes.string.isRequired,
  }),
  hostApi: PropTypes.shape({
    onClose: PropTypes.func,
  }).isRequired,
};
