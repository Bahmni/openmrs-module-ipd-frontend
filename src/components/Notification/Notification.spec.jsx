import React from "react";
import { render } from "@testing-library/react";
import Notification from "./Notification";

describe("Notification", () => {
  it("should show success notification", () => {
    const { container } = render(
      <Notification
        hostData={{ notificationKind: "success" }}
        hostApi={jest.fn()}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("should show warning notification", () => {
    const { container } = render(
      <Notification
        hostData={{ notificationKind: "warning" }}
        hostApi={jest.fn()}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
