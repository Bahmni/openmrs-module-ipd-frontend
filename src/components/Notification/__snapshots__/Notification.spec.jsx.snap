// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification should show success notification 1`] = `
<div>
  <div
    class="alertContainer"
  >
    <div>
      <div
        class="bx--inline-notification bx--inline-notification--low-contrast bx--inline-notification--success bx--inline-notification--hide-close-button"
        kind="success"
        role="alert"
      >
        <div
          class="bx--inline-notification__details"
        >
          <svg
            aria-hidden="true"
            class="bx--inline-notification__icon"
            fill="currentColor"
            focusable="false"
            height="20"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10,1c-4.9,0-9,4.1-9,9s4.1,9,9,9s9-4,9-9S15,1,10,1z M8.7,13.5l-3.2-3.2l1-1l2.2,2.2l4.8-4.8l1,1L8.7,13.5z"
            />
            <path
              d="M8.7,13.5l-3.2-3.2l1-1l2.2,2.2l4.8-4.8l1,1L8.7,13.5z"
              data-icon-path="inner-path"
              fill="none"
              opacity="0"
            />
            <title>
              success icon
            </title>
          </svg>
          <div
            class="bx--inline-notification__text-wrapper"
          >
            <p
              class="bx--inline-notification__title"
            >
              <div />
            </p>
            <div
              class="bx--inline-notification__subtitle"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Notification should show warning notification 1`] = `
<div>
  <div
    class="alertContainer"
  >
    <div>
      <div
        class="bx--inline-notification bx--inline-notification--low-contrast bx--inline-notification--warning bx--inline-notification--hide-close-button"
        kind="warning"
        role="alert"
      >
        <div
          class="bx--inline-notification__details"
        >
          <svg
            aria-hidden="true"
            class="bx--inline-notification__icon"
            fill="currentColor"
            focusable="false"
            height="20"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10,1c-5,0-9,4-9,9s4,9,9,9s9-4,9-9S15,1,10,1z M9.2,5h1.5v7H9.2V5z M10,16c-0.6,0-1-0.4-1-1s0.4-1,1-1	s1,0.4,1,1S10.6,16,10,16z"
            />
            <path
              d="M9.2,5h1.5v7H9.2V5z M10,16c-0.6,0-1-0.4-1-1s0.4-1,1-1s1,0.4,1,1S10.6,16,10,16z"
              data-icon-path="inner-path"
              opacity="0"
            />
            <title>
              warning icon
            </title>
          </svg>
          <div
            class="bx--inline-notification__text-wrapper"
          >
            <p
              class="bx--inline-notification__title"
            >
              <div />
            </p>
            <div
              class="bx--inline-notification__subtitle"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
